File: 卡码网/Js/test08.js
//8. 摆平积木
const rl=require("readline").createInterface({input:process.stdin});
const iter=rl[Symbol.asyncIterator]();
const readline=async ()=>(await iter.next()).value;

async function main(){
  while(line=await readline()){
    const n=parseInt(line);
    if(n===0) break;
    const nums=(await readline()).split(" ").map(Number);
    const total=nums.reduce((item,pre)=>item+pre,0);
    const avg=total/n;
    let res=0;
    nums.forEach(item=>{
      if(item>avg) res+=item-avg;
    });
    console.log(res);
    console.log();
  }
}

main();

File: 卡码网/Python/test08.py
#8. 摆平积木
import sys
from functools import reduce

def main():
	for line in sys.stdin:
		n=int(line)
		if n==0:
			break
		nums=list(map(int,sys.stdin.readline().split()))
		total=sum(nums)
		avg=total/n
		res=0
		for num in nums:
			if num>avg:
				res+=num-avg
		print(int(res))
		print()

if __name__ == '__main__':
	main()
